test/test_jit.py

```diff
@@ -5620,6 +5620,106 @@ def backward_hook(module, grad_input, grad_output):
         with self.assertRaisesRegex(Exception, "backward hooks assigned"):
             torch.jit.trace(n, (torch.tensor(1.0),))
 
+    def test_scripting_hooks(self):
+        class Net(nn.Module):
+            def forward(self, x):
+                return x + x
+
+        def forward_pre_hook(self, inputs):
+            # type: (Tuple[Tensor,]) -> Tuple[Tensor,]
```

Comment:
Just curious: so inputs / outputs should always be a tuple ? how does that work exactly here & in python ? I would kind of expect that for a single input/output you wouldn't need a tuple but otherwise you would.

Reply:
Although I am not so sure, this is what I believe is happening:
`__call__` method expects a `Tuple` parameter for `inputs`, we pass it to the forward-pre-hooks as is. These hooks can return a `Tuple` or not; if they don't return a `Tuple`, we create one inside the `__call__` method and continue. Depending on the hook implementation, I think this could also be `# type (Tuple[Tensor,]) ->Tensor`.